### DISCUSSION TITLE
Changing the dimension/shape of a tensor

### DISCUSSION THREAD

**Discussion by jiachenpauljiang** [mrdbourke/pytorch-deep-learning#45] [Q&A]
I have coded deep learning models in the past, but I never fully understood them. That's why I am learning it again from the scratch. I am about halfway down the first notebook, and this might be a really fundamental math question. When we change/reshape a tensor, say in order to do a matrix multiplication between two tensors with the same shape, the value of the tensor does not change although the dimension/shape has been changed. Is this correct? For example, the value of [[1,2,3], [1,2,3]] is the same as [[1,1], [2,2], [3,3]]? Thanks a lot!

**Comment by sriram-gupta** [comment]
what do you mean  when you say " the value doesn't change but the dimension changed ".  If you are talking in terms of what it represent it totally changes when we do any linear operation on it , be it transpose also.

**Comment by mrdbourke** [comment]
Yes @jiachenpauljiang ,  you're correct. 

You can reshape/tensor a tensor *without* changing the underlying values.

You are only rearranging the values or the axes/dimensions.

Rearranging tensors can help with various manipulations or calculations (e.g. matrix multiplication).
